### PR TITLE
Research API and Fix UI

### PR DESCRIPTION
**Pull request by ngochieu1297**
## [Pull request title]
Research API and Fix UI
### [Redmine link]
https://edu-redmine.framgia.vn/issues/7511

### CODE REVIEW

**Comment by ngochieu1297** [comment]
ready

File: app/src/main/java/com/example/music_58/data/source/remote/TrackAsyncTask.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TrackAsyncTask extends BaseAsyncTask<Track> {
+    private static final String ARTWORK_URL = " artwork_url";
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/data/source/remote/TrackAsyncTask.java]:
```suggestion
    private static final String ARTWORK_URL = "artwork_url";
```

File: app/src/main/java/com/example/music_58/data/source/remote/TrackAsyncTask.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private static final String COLLECTION = "collection";
+    private static final String ID = "id";
+    private static final String KEY_USER = "user";
+    private static final String KEY_USER_NAME = "username";
+    private static final String TITLE = " title";
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/data/source/remote/TrackAsyncTask.java]:
```suggestion
    private static final String TITLE = "title";
```

File: app/src/main/java/com/example/music_58/data/source/remote/TrackAsyncTask.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                int id = track.getInt(ID);
+                String title = track.getString(TITLE);
+                String artworkUrl = track.getString(ARTWORK_URL);
+                String artist = track.getJSONObject(KEY_USER).getString(KEY_USER_NAME);
+                Track object = new Track(id, title, artist);
+                tracks.add(object);
+            }
+        } catch (JSONException e) {
+            mException = e;
+            e.printStackTrace();
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/data/source/remote/TrackAsyncTask.java]: remove

File: app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+public class ViewPagerAdapter extends FragmentStatePagerAdapter {
+    public ViewPagerAdapter(FragmentManager fm) {
+        super(fm);
+    }
+
+    @Override
+    public Fragment getItem(int i) {
+        switch (i) {
+            case 0:
+                return new HomeFragment();
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java]: convention 2.11

File: app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            case 1:
+                return new FavoriteFragment();
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java]: tương tự

File: app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            case 2:
+                return new LocalMusicFragment();
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java]: tuương tự

File: app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    public ViewPagerAdapter(FragmentManager fm) {
+        super(fm);
+    }
+
+    @Override
+    public Fragment getItem(int i) {
+        switch (i) {
+            case 0:
+                return new HomeFragment();
+            case 1:
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/ui/adapter/ViewPagerAdapter.java]: case 0, 1, 2, 3 đang hardcode. em có thẻ đặt tên bằng dùng Anotation InDef: https://developer.android.com/reference/android/support/annotation/IntDef

**Comment by ngochieu1297** [comment]
ready

**Pull request merged**